src/Cli/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -50,9 +50,27 @@ public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSett
                             .EnumerateObject()
                             .FirstOrDefault(IsDefaultProfileType).Value;
                     }
-                    else
+                    else // Find a profile match for the given profileName
                     {
-                        if (!profilesObject.TryGetProperty(profileName, out profileObject) || profileObject.ValueKind != JsonValueKind.Object)
+                        IEnumerable<JsonProperty> caseInsensitiveProfileMatches = profilesObject
```

Comment:
It would be nice to refactor this into its own function, but that is a bit troublesome due to the required `out` parameter `profileObject`, potentially returning `null` or not for the launch settings application failure, and the fact that this interface is `static`. It turned out to be messier than not to have this code all-inlined.